lua/pulsar_lib/core/sql/sv_sql.lua

```diff
@@ -92,8 +127,63 @@ function SQL:RawQuery(query, onSuccess, onError)
 
 		local x = string.Split(query, "\n")
 		for _, line in ipairs(x) do
-			PulsarStore.Logging.Debug(line)
+			PulsarLib.Logging.Debug(line)
+		end
+		query = sql.Query(query)
+
+		if query == false then
+			PulsarLib.Logging.Fatal("SQL query failed!")
+			PulsarLib.Logging.Fatal(sql.LastError())
+			onError(sql.LastError())
+		else
+			onSuccess(query)
+		end
+	end
+end
+
+function SQL:PreparedQuery(query, values, onSuccess, onError)
+	onSuccess = onSuccess or emptyFunction
+	onError = onError or emptyFunction
+
+	if self.Details.UsingMySQL then
+		local queryObj = self.Connection:prepare(query)
+
+		queryObj.onSuccess = function(_, data)
+			onSuccess(data)
+		end
+
+		queryObj.onError = function(_, err)
+			PulsarLib.Logging.Fatal("Prepared MySQL query failed!")
+			PulsarLib.Logging.Fatal(err)
+			PulsarLib.Logging.Fatal(SQL:prepareStatement(query, values))
+			onError(err)
+		end
+
+		for k, v in ipairs(values or {}) do
+			if type(v) == "string" then
+				queryObj:setString(k, v)
+			elseif type(v) == "number" then
+				queryObj:setNumber(k, v)
+			elseif type(v) == "boolean" then
+				queryObj:setBoolean(k, v)
+			elseif v == nil then
+				queryObj:setNull(k)
+			end
+		end
+
+		queryObj:start()
+	else
+		for k, v in pairs(sqliteReplaces) do
+			query = string.Replace(query, k, v)
+		end
+
+		local x = string.Split(query, "\n")
+		for _, line in ipairs(x) do
+			PulsarLib.Logging.Debug(line)
```

Comment:
This should probably be at one of the trace levels.

Reply:
i don't understand what you mean with that, Debug is already very low and there are no Trace methods

Reply:
Oh, is this the build with trace level, but no trace methods? I'll fix that later.